tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/WeightedSnapshotWithExemplars.java

```diff
@@ -70,22 +71,21 @@ Object exemplarMetadata() {
      * @param values an unordered set of values in the reservoir
      */
     WeightedSnapshotWithExemplars(ExemplarMetadataProvider<?> provider, Collection<WeightedSampleWithExemplar> values) {
-        ImmutableList.Builder<WeightedSample> weightedSamplesBuilder =
-                ImmutableList.builderWithExpectedSize(values.size());
-        ImmutableList.Builder<LongExemplar<Object>> exemplarsBuilder = null;
+        List<WeightedSample> weightedSamplesBuilder = new ArrayList<>(values.size());
+        List<LongExemplar<Object>> exemplarsBuilder = null;
 
         for (WeightedSampleWithExemplar v : values) {
             weightedSamplesBuilder.add(new WeightedSample(v.value, v.weight));
             if (v.exemplarMetadata != null) {
                 if (exemplarsBuilder == null) {
-                    exemplarsBuilder = ImmutableList.builder();
+                    exemplarsBuilder = new ArrayList<>();
                 }
                 exemplarsBuilder.add(DefaultLongExemplar.of(v.exemplarMetadata, v.value));
             }
         }
-        this.exemplars = (exemplarsBuilder == null) ? ImmutableList.of() : exemplarsBuilder.build();
+        this.exemplars = (exemplarsBuilder == null) ? List.of() : Collections.unmodifiableList(exemplarsBuilder);
```

Comment:
Since we are always setting `this.exemplars` to _some_ list, can we set `exemplarsBuilder` to an empty collection instead of `null` and then avoid the `if (exemplarsBuilder == null)` check and also simplify this line to just `Collections.unmodifiableList(exemplarsBuilder);` 

What benefit are we getting by having the builder be `null`? At the start of the method?

Reply:
I was trying to avoid an ArrayList allocation, but we can just eat it (empty `ArrayList` doesn't allocate the backing `Object[]` array until an element is added, so its not too bad.